blockchain/chain/src/store/chain_store.rs

```diff
@@ -494,6 +494,78 @@ where
         Ok(miner::State::load(self.blockstore(), &actor)?)
     }
 
+    /// Imports a StateTree given an AsyncRead that has data in CAR format
+    pub async fn import_state_tree<R: AsyncRead + Send + Unpin>(
+        &self,
+        reader: R,
+    ) -> Result<Cid, Error> {
+        let state_root = forest_car::load_car(self.blockstore(), reader)
+            .await
+            .map_err(|e| Error::Other(format!("Import StateTree failed: {}", e.to_string())))?;
+        if state_root.len() != 1 {
```

Comment:
You could use [pop](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.pop) and if it returns a `None` variant return the error. This will get rid of indexing.